sdk/typescript/src/client/cryptography.ts

```diff
@@ -372,6 +372,51 @@ export function publicKeyFromDWalletOutput(dWalletOutput: Uint8Array): Uint8Arra
 	return Uint8Array.from(public_key_from_dwallet_output(dWalletOutput));
 }
 
+/**
+ * Verify the user's public output and get the DKG public output.
+ *
+ * @param dWallet - The DWallet object containing the user's public output
+ * @param encryptedUserSecretKeyShare - The encrypted user secret key share
+ * @param publicKey - The user's public key, right now only SECP256K1 is supported.
+ * @returns The DKG public output
+ */
+export async function verifyAndGetDWalletDKGPublicOutput(
+	dWallet: DWallet,
+	encryptedUserSecretKeyShare: EncryptedUserSecretKeyShare,
+	publicKey: PublicKey,
+): Promise<Uint8Array> {
+	if (
+		SIGNATURE_FLAG_TO_SCHEME[publicKey.flag() as keyof typeof SIGNATURE_FLAG_TO_SCHEME] !==
+		'Secp256k1'
+	) {
+		throw new Error('Only Secp256k1 keypairs are supported for now');
+	}
+
+	if (!dWallet.state.Active?.public_output) {
+		throw new Error('DWallet is not in active state');
+	}
+
+	if (!encryptedUserSecretKeyShare.state.KeyHolderSigned?.user_output_signature) {
+		throw new Error('User output signature is undefined');
+	}
+
+	const userPublicOutput = Uint8Array.from(dWallet.state.Active.public_output);
```

Comment:
this is not the user public output, only network public outputs are saved on chain